GitTfs/Core/Changes/Git/RenameEdit.cs

```diff
@@ -6,24 +7,100 @@ public class RenameEdit : IGitChangedFile
         public string Path { get; private set; }
         public string PathTo { get; private set; }
         public string NewSha { get; private set; }
+        public string OldSha { get; private set; }
         public string Score { get; private set; }
         public IGitRepository _repository { get; private set; }
 
         public RenameEdit(IGitRepository repository, GitChangeInfo changeInfo)
         {
             _repository = repository;
             NewSha = changeInfo.newSha;
+            OldSha = changeInfo.oldSha;
             Path = changeInfo.path;
             PathTo = changeInfo.pathTo;
             Score = changeInfo.score;
         }
 
         public void Apply(ITfsWorkspaceModifier workspace)
         {
-            workspace.Edit(Path);
-            workspace.Rename(Path, PathTo, Score);
-            var workspaceFile = workspace.GetLocalPath(PathTo);
-            _repository.CopyBlob(NewSha, workspaceFile);
+            if (Path.ToLower() != PathTo.ToLower())
+            {
+                var workspaceFile = workspace.GetLocalPath(PathTo);
+                workspace.Edit(Path);
+                workspace.Rename(Path, PathTo, Score);
+                _repository.CopyBlob(NewSha, workspaceFile);
+            }
+            else
+            {
+                //This change can not be mapped to TFS, since TFS is case insensitive
+                var workspaceFile = workspace.GetLocalPath(Path);
```

Comment:
Sorry but I don't understand exactly what we try to achieve here (it should be my fault because I was not there at the beginning).

I understand that we compare the content of 2 blobs but what does the content of the 2 blobs represent exactly?
What is `changeInfo` ?

Because I see that you use `changeInfo.newSha` and `changeInfo.oldSha` to name the files and because that's `sha`, these 2 values will be the same if the content is the same and different if the content is different!

So, it seems that we don't need to compare the content, no? Does I miss something?